src/main/java/com/google/sps/image/ImageScorer.java

```diff
@@ -35,6 +37,15 @@ public ImageScorer(String[] keywords) {
     this.keywords = keywords;
   }
 
+  private Map<String, Boolean> generateMap(List<String> elements) {
```

Comment:
Why can't you just build a set

`new HashSet(elements)`

and be done with it?

https://docs.oracle.com/javase/7/docs/api/java/util/HashSet.html#HashSet(java.util.Collection)

Or am I missing something here?

Reply:
That would work with HashSet, but I used a HashMap because it has O(1) time complexity for searching a key.
HashMap does not have java.util.Collection as constructor argument.

Reply:
How large are N and M (as described in PR description)? Would it even matter?

Reply:
N = [2,3] and M = [2, 15]

Reply:
With N*M = 30, this optimization is purely theoretical, you would not notice any difference. 

Although reducing the number of queries (or parallelizing them) to search engine will make a difference in waiting time.